lib/modules/asset/AssetsGroupsController.ts

```diff
@@ -159,12 +171,60 @@ export class AssetsGroupsController {
     }
   }
 
+  async checkGroupName(
+    engineId: string,
+    body: AssetsGroupsBodyRequest,
+    assetId?: string
+  ) {
+    if (typeof body.name !== "string") {
+      return;
+    }
+
+    const groupsCount = await this.sdk.document.count(
+      engineId,
+      InternalCollection.ASSETS_GROUPS,
+      {
+        query: {
+          bool: {
+            must: [
+              {
+                regexp: {
+                  name: {
+                    case_insensitive: true,
+                    value: body.name,
+                  },
+                },
+              },
+            ],
+            must_not: [
+              {
+                terms: {
+                  _id: typeof assetId === "string" ? [assetId] : [],
+                },
+              },
+            ],
+          },
+        },
```

Comment:
I didn't manage to use regexp in case-insensitive mode in Koncorde, the flag `i` like it's explain in documentation throw an error and like I haven't the time to search more at the moment I have preferred to use Elasticsearch syntax that works correctly.

But it's true, maybe we need to refactor that later

Reply:
Indeed... Weird bug. I suggest adding a comment above :shrug:

Reply:
I dug a little bit and found, that atm, `regexp` is not usable in search queries. Because it's not translated from Koncorde DSL to ES DSL (Idk why).

So if we don't implement it (because there are maybe good reasons :shrug:), you can either add a comment to explain why we use a ES query, or I also suggest to use the `regexp` syntax from ES (yes, therefore you can mix koncorde & es lol):
```ts
query: {
  and: [
    {
      regexp: {
        name: {
          case_insensitive: true,
          value: body.name,
        },
      },
    },
    {
      not: {
        "equals": { _id: typeof assetId === "string" ? [assetId] : [] }
      }
    }
  ],
},
```

Reply:
Koncorde regex and ES regex does not have the same syntax that's why

Reply:
I don't understand. Actually the keyword is allowed when searching for documents, via Koncorde.

Reply:
Koncorde requests are supposed to be used indifferently in realtime filters and in search queries but if you use a regex then it cannot works in both because the syntax is difference